pallets/interest-accrual/src/lib.rs

```diff
@@ -241,47 +263,40 @@ pub mod pallet {
 			interest_rate_per_sec: T::InterestRate,
 			normalized_debt: T::Balance,
 		) -> Result<T::Balance, DispatchError> {
-			Rate::<T>::get(interest_rate_per_sec)
-				.ok_or(Error::<T>::NoSuchRate)
-				.and_then(|rate| {
-					Self::calculate_debt(normalized_debt, rate.accumulated_rate)
-						.ok_or(Error::<T>::DebtCalculationFailed)
-				})
-				.map_err(Into::into)
+			let rate = Self::get_rate(interest_rate_per_sec)?;
+			let debt = Self::calculate_debt(normalized_debt, rate.accumulated_rate)
+				.ok_or(Error::<T>::DebtCalculationFailed)?;
+			Ok(debt)
```

Comment:
Could: End a bit shorter
```suggestion
			Self::calculate_debt(normalized_debt, rate.accumulated_rate)
				.ok_or(Error::<T>::DebtCalculationFailed.into())
```

Reply:
I went with the extra line so I could let the question mark operator do my error conversion, avoiding a manual `.into()`.